### PR TITLE
User can search for a song to post

### PR DESCRIPTION
**Pull request by salindholm**
[User can search for a song to post](https://www.pivotaltracker.com/story/show/176623860)

## User Story
```
As a User,            
In order to find a specific song to share,         
I want to be able to search for a song.
```
## Changes proposed in this pull request:
* Creating PostForm component 
* Adding Tab navigator in App component with search input and button
* Adding onChangeText to search input and button
* Creating TrackService module to fetch search results from API
* Adding action for SET_SEARCH_INDEX and adding initialState to search result as an empty array
* Creating test with fixture file

## What I have learned working on this feature:
* How to write component tests with Jest in React Native
* How to write feature tests with Cypress in React Native
* How to work with FlatLists

## Packages/Gems added
* @react-navigation/bottom-tabs

## Screenshots (Client)
Screen Shot 2021-01-28 at 11 49 27 AM
Screen Shot 2021-01-28 at 11 53 13 AM

### CODE REVIEW

File: App.jsx [hunk 1]
```diff
@@ -4,25 +4,39 @@ import { createStackNavigator } from '@react-navigation/stack'
...
+    </Stack.Navigator>
+  )
+}
+
+const App = () => {
+  return (
+    <>
+      <NavigationContainer style={{ height: 10 }}>
+        <Tab.Navigator>
+          <Tab.Screen name='Feed' component={HomeStack} />
```
Comment (ViktorHek) [inline App.jsx:35]: if you are importing the `HomeStack` (i would call it HomeScreen but it doesn't matter) then I would suggest putting the `HomeStack` in a separate component.

File: index.js [hunk 2]
```diff
@@ -4,6 +4,8 @@ import { Provider } from 'react-redux'
 import store from './src/state/store/store'
 import { registerRootComponent } from 'expo'
 
+global.store = store
```
Comment (ViktorHek) [inline index.js:7]: what is this??

File: App.jsx [hunk 3]
```diff
@@ -4,25 +4,39 @@ import { createStackNavigator } from '@react-navigation/stack'
...
+          <Tab.Screen name='Post' component={PostForm} />
+        </Tab.Navigator>
+      </NavigationContainer>
+    </>
   )
```
Comment (sealfury) [inline App.jsx:40]: Good idea with `HomeStack` component for tab navigation, but we should probably extract it out of the `App` component

File: specs/PostForm.spec.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+
+    it('contains search button', () => {
+      expect(screen.getByTestId('searchButton')).toBeTruthy()
+    })
+
+    it('contains search results', () => {
+      expect(screen.getByTestId('searchResults')).toBeTruthy()
+    })
+  })
+})
```
Comment (sealfury) [inline specs/PostForm.spec.js]: obviously not ideal to only use these assertions but I appreciate that cypress picks up the slack on that :)

File: src/styles/styles.js [hunk 5]
```diff
@@ -29,33 +30,33 @@ const styles = StyleSheet.create({
...
-    color: 'aliceblue',
+    color: 'grey',
     opacity: 0.6,
     fontSize: 70,
     alignSelf: 'center',
     marginTop: 40,
   },
   stopButton: {
-    color: 'lavenderblush',
+    color: 'grey',
```
Comment (sealfury) [inline src/styles/styles.js:59]: can't say i'm thrilled about these colour changes but thanks for taking care of the fonts

File: src/components/PostForm.jsx [hunk 6]
```diff
@@ -0,0 +1,39 @@
...
+      />
+      <Button
+        testID='searchButton'
+        title='Search'
+        onPress={() => TrackService.index(search)}
+      />
+      <FlatList
+        testID='searchResults'
+        data={searchResult}
+        keyExtractor={(item, index) => index.toString()}
```
Comment (sealfury) [inline src/components/PostForm.jsx:27]: nice work on the extraction!

File: package.json [hunk 7]
```diff
@@ -11,8 +11,9 @@
     "test": "jest --rootDir=specs --verbose"
   },
   "dependencies": {
+    "@react-native-community/masked-view": "^0.1.10",
```
Comment (sealfury) [inline package.json]: can you explain why we're using the major version of this dependency now?  I assume it was intentional but if not i always worry about automatic updates clashing with other dependencies down the road...

File: index.js (hunk 2 above)
Comment (salindholm) [reply index.js:7]: That is equivalent to window.store in React!

File: App.jsx (hunk 1 above)
Comment (KyunginNa) [reply App.jsx:35]: We might want to use appTitle and Stack.Navigator for other components later, so we'll keep it for now. But yea as you said, we can definitely extract it later!👍

File: package.json (hunk 7 above)
Comment (KyunginNa) [reply package.json]: As you said, we will delete `^`! Thanks👍

**Pull request merged**